lib/linguist/heuristics.yml

```diff
@@ -567,6 +567,18 @@ disambiguations:
   rules:
   - language: XML
     pattern: '<TS\b'
+  - language: AssemblyScript
+    negative_pattern: 
+    - '\s+undefined\s+'
+    - '\s+any\s+'
+    - '\s+unknown\s+'
```

Comment:
Btw I'm not sure about `unknown`. It is quite likely that it will still be used at least for variadic function decladarions

Reply:
Also some people in the AS community have implemented various forms of `unknown`, and it could conceivably be supported officially later once semantics are ironed out.